org.thecrossingchurch.CustomJobs/Jobs/HubspotIntegrationMatching.cs

```diff
@@ -0,0 +1,718 @@
+            int idSet = 0;
+            // HS person with a person id that no longer exists in Rock
+            int idRemoved = 0;
+            // HS person with a person id that exists in Rock today
+            int idCorrect = 0;
+            // HS person has no match in rock
+            int idEmpty = 0;
+            // Inserted new HS person that was only in rock
+            int idAdded = 0;
+
+            
+
+            // Process each contact in HubSpot
+            for ( var i = 0; i < contacts.Count(); i++ )
+            {
+                var contact = contacts[i];
+                
+                Stopwatch watch = new Stopwatch();
+                watch.Start();
+
+                Debug.WriteLine("Operating on contact: " + contact);
+
+                if ( contact.properties.rock_person_id != "" )
+                {
+                    Person existingPerson = personService.Get(contacts[i].properties.rock_person_id);
+                    if (existingPerson == null) // HS Contact person does not exist in Rock
+                    {
+                        // Person not found
+                        Debug.WriteLine("Contact with id not found in Rock. Clearing Id.");
+                        // Clear the person id in HubSpot
+                        var properties = new List<HubspotPropertyUpdate>() { new HubspotPropertyUpdate() { property = "rock_person_id", value = "" } };
+                        var url = $"https://api.hubapi.com/crm/v3/objects/contacts/{contact.id}";
+                        UpdateHsContact(current_id, url, properties, 0);
+                        idRemoved++;
+                    }
+                    else
+                    {
+                        // Person still exists in DB
+                        // No need to continue processing this person
+                        Debug.WriteLine("Contact with id found in Rock. Skipping contact");
+                        idCorrect++;
+                        break;
+                    }
+                } 
+                
+                Person person = null;
+                bool hasMultiEmail = false;
+                
+                List<int> matchingIds = FindPersonIds(contact, _context);
+                if ( matchingIds.Count > 1 )
+                {
+                    hasMultiEmail = true;
+                }
+
+                if ( matchingIds.Count == 1 )
+                {
+                    person = personService.Get(matchingIds.First());
+                    // New single match found, increment the counter!
+                    idSet++;
+                }
+
+                Debug.WriteLine("Matching People Found: " + matchingIds.ToString());
+                WriteToLog(string.Format("    After SQL: {0}", watch.ElapsedMilliseconds));
+
+                // Atempt to match 1:1 based on email history making sure we exclude emails with multiple matches in the person table
+                if ( person == null && !hasMultiEmail )
```

Comment:
I've mentioned this in a previous comment, but I'd prefer `hasMultiEmail == false` over `!hasMultiEmail`. Spelling out the `== false` is more clear, which would make it easier to read later.